### PR TITLE
Add signatures to the certificates.

### LINKED ISSUE #578: stake registration/delegation/.. certificate are missing their own signatures

### PR DESCRIPTION
**Pull request by qnikst**
closes #578

### CODE REVIEW

**Comment by NicolasDP** [comment]
@qnikst , I believe you are a vim user, right? https://github.com/rust-lang/rust.vim#formatting-with-rustfmt

**Comment by qnikst** [comment]
@vincenthz I've updated the code for everything except for `CertificateContent::StakePoolRetirement` will work on that now, I have not introduced generic code, though it can save a couple of lines.

File: chain-impl-mockchain/src/certificate.rs [hunk 1]
```diff
@@ -209,37 +290,56 @@ impl Readable for StakeDelegation {
...
+    }
+
+    pub fn verify(&self, raw_signatures: &[SignatureRaw]) -> Verification {
+        use crate::key::{deserialize_signature, verify_signature};
+        let mut owners = self.owners.iter();
+        let mut signatures = raw_signatures.iter();
+        loop {
+            match owners.next() {
+                None => return Verification::Success,
+                Some(owner) => match signatures.next() {
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/certificate.rs]: or:

```rust
for owner in owners {
}
```

**Comment by qnikst** [comment]
@vincenthz I've updated pool rederegistration support and introduced a common trait for certificates loading. @NicolasDP rustfmt included!

**Pull request merged**